flang/lib/Optimizer/HLFIR/Transforms/BufferizeHLFIR.cpp

```diff
@@ -359,6 +359,7 @@ struct GetLengthOpConversion
     fir::FirOpBuilder builder(rewriter, getLength.getOperation());
     hlfir::Entity bufferizedExpr = getBufferizedExprStorage(adaptor.getExpr());
     mlir::Value length = hlfir::genCharLength(loc, builder, bufferizedExpr);
+    length = builder.createConvert(loc, builder.getIndexType(), length);
     if (!length)
```

Comment:
I think you need to move the change to after the `if`. Otherwise it will try to use the empty value when building the convert.